dallinger/bots.py

```diff
@@ -231,7 +232,12 @@ def sign_up(self):
                     hash=uuid.uuid4().hex
                 )
             )
-            result = requests.post(url)
+            try:
+                result = requests.post(url)
+            except ConnectionError:
```

Comment:
I'd suggest maybe using `result.raise_for_status()` here to combine the two calls to `self.stochastic_sleep()` into one exception handler with `requests.exceptions.RequestException`. That wouldn't handle the `json()[' status'] == 'error'` case though, but that case could do an explicit `raise requests.exceptions.HTTPError('JSON error status returned: {}'.format(result.json()))`. Then maybe add a `self.log("Bot request exception, retrying: {}".format(e))` in the exception handler to ensure we get information about these failures.

Reply:
I feel a bit off about raising an explicit HTTP error here considering we're not putting it in the status. I think I'd rather have the 2 line duplication than the unusual raise. I think you're absolutely right about the `raise_for_status()` though, so I'm going to incorporate that.